relay_messages_test.go

```diff
@@ -161,58 +162,73 @@ func withLazyCallReqCombinations(f func(cr testCallReq)) {
 type testCallRes int
 
 type testCallResParams struct {
-	payloadSize       int
 	hasFragmentedArg2 bool
 
-	flags    byte
-	code     byte
-	span     [25]byte
-	headers  map[string]string
-	csumType byte
-	arg1     []byte
-	arg2     []byte
-	arg3     []byte
+	flags       byte
+	code        byte
+	span        [25]byte
+	headers     map[string]string
+	csumType    byte
+	arg1        []byte
+	arg2Prefix  []byte // used for corrupting arg2
+	arg2KeyVals map[string]string
+	arg3        []byte
 }
 
 const (
 	resIsContinued testCallRes = (1 << iota)
 	resIsOK
 	resHasHeaders
 	resHasChecksum
+	resHasArg2
+	resHasFragmentedArg2
 	resTotalCombinations
 )
 
 func (cr testCallRes) res(tb testing.TB) lazyCallRes {
-	params := testCallResParams{
-		payloadSize: 100,
+	var params testCallResParams
+
+	if cr&resHasFragmentedArg2 != 0 {
+		params.hasFragmentedArg2 = true
 	}
-	if cr&resIsContinued != 0 {
+	if cr&(resIsContinued|resHasFragmentedArg2) != 0 {
 		params.flags |= hasMoreFragmentsFlag
 	}
 	if cr&resIsOK == 0 {
 		params.code = 1
 	}
+	params.headers = map[string]string{}
 	if cr&resHasHeaders != 0 {
-		params.headers = map[string]string{
-			"k1":      "v1",
-			"k222222": "",
-			"k3":      "thisisalonglongkey",
-		}
+		params.headers["k1"] = "v1"
+		params.headers["k222222"] = ""
+		params.headers["k3"] = "thisisalonglongkey"
+	}
+	if cr&(resHasArg2|resHasFragmentedArg2) != 0 {
```

Comment:
why do we check `HasFragmentedArg2` here?

Reply:
so the arg scheme is set to `thrift`. we only support arg2 iteration for `thrfit`, and the two types here are meant for adding key/vals to arg2

Reply:
ah, would it be clearer to use a separate orthogonal bit like `resIsThrift` since it is possible to have a fragmented arg2 without thrift (though we may not necessarily find it useful in our tests)